dali/imgcodec/decoders/nvjpeg/nvjpeg_helper.h

```diff
@@ -0,0 +1,245 @@
+        return "The library handle was not initialized.";
+      case NVJPEG_STATUS_INVALID_PARAMETER:
+        return "Wrong parameter was passed. For example, a null pointer as input data, or "
+               "an image index not in the allowed range.";
+      case NVJPEG_STATUS_BAD_JPEG:
+        return "Cannot parse the JPEG stream. Check that the encoded JPEG stream and its "
+               "size parameters are correct.";
+      case NVJPEG_STATUS_JPEG_NOT_SUPPORTED:
+        return "Attempting to decode a JPEG stream that is not supported by the nvJPEG library.";
+      case NVJPEG_STATUS_ALLOCATOR_FAILURE:
+        return "The user-provided allocator functions, for either memory allocation or for "
+               "releasing the memory, returned a non-zero code.";
+      case NVJPEG_STATUS_EXECUTION_FAILED:
+        return "Error during the execution of the device tasks.";
+      case NVJPEG_STATUS_ARCH_MISMATCH:
+        return "The device capabilities are not enough for the set of input parameters "
+               "provided (input parameters such as backend, encoded stream parameters, "
+               "output format).";
+      case NVJPEG_STATUS_INTERNAL_ERROR:
+        return "Error during the execution of the device tasks.";
+      case NVJPEG_STATUS_IMPLEMENTATION_NOT_SUPPORTED:
+        return "Not supported.";
+#if NVJPEG_VER_MAJOR > 11 || \
+    (NVJPEG_VER_MAJOR == 11 && NVJPEG_VER_MINOR >= 6)
+      case NVJPEG_STATUS_INCOMPLETE_BITSTREAM :
+        return "Bitstream input data incomplete.";
+#endif
+      default:
+        return "< unknown error >";
+    }
+  }
+
+  static std::string Message(nvjpegStatus_t result, const char *details) {
+    if (details && *details) {
+      return make_string("nvJPEG error (", result, "): ", ErrorString(result),
+                         "\nDetails:\n", details);
+    } else {
+      return make_string("nvJPEG error (", result, "): ", ErrorString(result));
+    }
+  }
+
+
+  nvjpegStatus_t result() const { return result_; }
+
+ private:
+  nvjpegStatus_t result_;
+};
+
+// Obtain nvJPEG library version or -1 if it is not available
+int nvjpegGetVersion();
+
+struct StateNvJPEG {
+  nvjpegBackend_t nvjpeg_backend;
+  nvjpegBufferPinned_t pinned_buffer;
+  nvjpegJpegState_t decoder_host_state;
+  nvjpegJpegState_t decoder_hybrid_state;
+  nvjpegJpegStream_t jpeg_stream;
+};
+
+template <typename T>
+struct EncodedImageInfo {
+  bool nvjpeg_support;
+  T c;
+  T widths[NVJPEG_MAX_COMPONENT];
+  T heights[NVJPEG_MAX_COMPONENT];
+  nvjpegChromaSubsampling_t subsampling;
+  CropWindow crop_window;
+};
+
+inline nvjpegJpegState_t GetNvjpegState(const StateNvJPEG& state) {
+  switch (state.nvjpeg_backend) {
+    case NVJPEG_BACKEND_HYBRID:
+      return state.decoder_host_state;
+    case NVJPEG_BACKEND_GPU_HYBRID:
+      return state.decoder_hybrid_state;
+    default:
+      DALI_FAIL("Unknown nvjpegBackend_t "
+                + std::to_string(state.nvjpeg_backend));
+  }
+}
```

Comment:
Not used - is it used later (with HW decoder)?

Reply:
No, it was removed in 8e45c0fbc6f0820c94ad2798cff516fd5e14dc90